src/Symfony/Component/Cache/Adapter/AbstractAdapter.php

```diff
@@ -108,7 +109,9 @@ public static function createSystemCache($namespace, $defaultLifetime, $version,
         }
 
         $apcu = new ApcuAdapter($namespace, (int) $defaultLifetime / 5, $version);
-        if (null !== $logger) {
+        if ('cli' === PHP_SAPI && !ini_get('apc.enable_cli')) {
+            $apcu->setLogger(new NullLogger());
+        } elseif (null !== $logger) {
```

Comment:
Why do we pass the `NullLogger` here? Why do we just not call `setLogger()` instead?

Like this:

```php
if (null !== $logger && ('cli' !== PHP_SAPI || ini_get('apc.enable_cli'))) {
    // ...
}
```